test/cbmc/proofs/ota_interface/setDataInterface/setDataInterface_harness.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * AWS IoT Over-the-air Update v3.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file setDataInterface_harness.c
+ * @brief Implements the proof harness for setDataInterface function.
+ */
+/* Ota interface includes. */
+#include "ota_interface_private.h"
+
+void setDataInterface_harness()
+{
+    OtaDataInterface_t * pDataInterface;
+    OtaDataInterface_t dataInterface;
+    OtaErr_t err;
+
+    uint8_t * pProtocol;
+    size_t size;
+
+    /* Initialize the variable. */
+    pProtocol = ( uint8_t * ) malloc( size );
```

Comment:
How does a malloc with a variable that's not set work with these proofs? Is there an upper limit on what size could be? If the value can be the max size of an int, this could cause some issues running the test.

Reply:
Yes, there is a upper limit on the `size` variable. I have added an assert statement to constrain this value from going above the upper limit.